### PR TITLE
Endpoint to Download Clinical Data for Files

### PR DESCRIPTION
**Pull request by joneubank**
## Description of Changes

New endpoint added at `POST /clinical/api/donors/data-for-files` which accepts a list of File IDs and will return the clinical data for the Donor(s) associated with those files. The Gateway will perform several permission checks before requesting the required data from the Clinical API:

- Requested files need to be listed as `has_clinical_data` in the file centric index.
- User must have permission to download each of the requested files, Embargo rules are enforced.
- User must have DACO access permission

The endpoint will return the download of a zip file containing all clinical entity TSVs relating to the donors for the requested files. This is the same TSV bundle that we generate for program data download for program submitters.

### Type of Change

- [ ] Bug
- [x] New Feature

## Breaking Changes
- The existing clinical proxy endpoint URLs have been modified and will need to be updated where used in the platform UI. The paths used to be `/clinical/{{path}}` and need to be updated to be `/clinical/proxy/{{path}}`

### CODE REVIEW

**Comment by joneubank** [comment]
Converted to draft so that this doesn't get merged before the UI has changes made to acommodate breaking changes.

File: .eslintrc.js [hunk 1]
```diff
@@ -26,6 +26,12 @@ module.exports = {
 			},
 		],
 		'@typescript-eslint/no-explicit-any': 'off',
+		'@typescript-eslint/no-unused-vars': [
+			'warn', // or "error"
+			{
+				argsIgnorePattern: '^_',
```
Comment (justincorrigible) [inline .eslintrc.js:32]: this thing is winking 😉 

(nice rule change, btw)

File: src/app.ts [hunk 2]
```diff
@@ -166,7 +168,7 @@ const init = async () => {
 
 	// Routers
 	app.use('/kafka', createKafkaRouter(egoClient));
-	app.use('/clinical', clinicalProxyRoute);
+	app.use('/clinical', await createClinicalRouter(esClient, egoClient));
 	app.use('/file-centric-tsv', await createFileCentricTsvRoute(esClient, egoClient));
```
Comment (justincorrigible) [inline src/app.ts:172]: is the `await` necessary here?

Comment (joneubank) [reply src/app.ts:172]: yes

File: src/config.ts [hunk 3]
```diff
@@ -54,6 +54,12 @@ export const EGO_VAULT_SCORE_PROXY_SECRET_PATH = process.env
 export const EGO_SCORE_PROXY_CLIENT_ID = process.env.EGO_SCORE_PROXY_CLIENT_ID;
 export const EGO_SCORE_PROXY_CLIENT_SECRET = process.env.EGO_SCORE_PROXY_CLIENT_SECRET;
 
+// Ego Credentials for Clinical API
+export const EGO_VAULT_CLINICAL_API_SECRET_PATH = process.env
+	.EGO_VAULT_CLINICAL_API_SECRET_PATH as string;
```
Comment (justincorrigible) [inline src/config.ts:59]: what makes this assertion necessary?

**Review by justincorrigible — APPROVED** [review]

**Comment by joneubank** [comment]
Ready for review. Platform-UI relevant changes here: https://github.com/icgc-argo/platform-ui/pull/2715

**Pull request merged**